scipy/optimize/optimize.py

```diff
@@ -1088,6 +1088,41 @@ def fmin_bfgs(f, x0, fprime=None, args=(), gtol=1e-5, norm=Inf,
     allvecs  :  list
         The value of xopt at each iteration. Only returned if retall is True.
 
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from scipy.optimize import fmin_bfgs
+
+    >>> def rosenbrock(x):
+    ...     # Rosenbrock Function,
+    ...     # which has a global minimum 0 at (1, 1)
+    ...     # https://en.wikipedia.org/wiki/Rosenbrock_function
+    ...     return (1-x[0])**2 + 100*(x[1]-x[0]**2)**2
+
+    >>> x0 = np.array([-3, -4])
+    >>> fmin_bfgs(rosenbrock, x0)
+    Warning: Desired error not necessarily achieved due to precision loss.
+            Current function value: 0.000000
+            Iterations: 64
+            Function evaluations: 330
+            Gradient evaluations: 106
+    array([0.99999669, 0.99999329])
```

Comment:
Looks like the CI refguide (documentation check) failure is real because of a small difference:

```
2021-05-16T16:37:56.5086553Z scipy.optimize.fmin_bfgs
2021-05-16T16:37:56.5088080Z ------------------------
2021-05-16T16:37:56.5088624Z 
2021-05-16T16:37:56.5089732Z File "build/testenv/lib/python3.8/site-packages/scipy/optimize/optimize.py", line 1103, in fmin_bfgs
2021-05-16T16:37:56.5090575Z Failed example:
2021-05-16T16:37:56.5091325Z     fmin_bfgs(rosenbrock, x0)
2021-05-16T16:37:56.5092012Z Expected:
2021-05-16T16:37:56.5092758Z     Warning: Desired error not necessarily achieved due to precision loss.
2021-05-16T16:37:56.5093610Z             Current function value: 0.000000
2021-05-16T16:37:56.5094354Z             Iterations: 64
2021-05-16T16:37:56.5095095Z             Function evaluations: 330
2021-05-16T16:37:56.5095858Z             Gradient evaluations: 106
2021-05-16T16:37:56.5096580Z     array([0.99999669, 0.99999329])
2021-05-16T16:37:56.5097249Z Got:
2021-05-16T16:37:56.5097913Z     Optimization terminated successfully.
2021-05-16T16:37:56.5098687Z              Current function value: 0.000000
2021-05-16T16:37:56.5099437Z              Iterations: 65
2021-05-16T16:37:56.5100163Z              Function evaluations: 246
2021-05-16T16:37:56.5100920Z              Gradient evaluations: 82
2021-05-16T16:37:56.5101644Z     array([0.99999232, 0.99998463])
```

Reply:
Could you elaborate what I can do for this failure?
(I'm sorry about my limit knowledge of doctest and if I'm not following some contributor guideline)

I'm a bit confused because I got the following varieties of outputs on
Windows10, Conda, Python 3.8.5 amd64, numpy 1.20.1, scipy 1.6.2&1.6.3 
```
Warning: Desired error not necessarily achieved due to precision loss.
         Current function value: 0.000000
         Iterations: 64
         Function evaluations: 330
         Gradient evaluations: 106
array([0.99999669, 0.99999329])
```

Win10, WSL2 Ubuntu20.04, Python 3.8.5, numpy 1.20.3, scipy 1.6.3 
```
Optimization terminated successfully.
         Current function value: 0.000000
         Iterations: 66
         Function evaluations: 246
         Gradient evaluations: 82
array([0.99999551, 0.99999101])
````

[Probably outdated] Win10, WSL2 Ubuntu18.04, Python 3.6.5, numpy 1.18.1, scipy 1.5.4 
```
Optimization terminated successfully.
         Current function value: 0.000000
         Iterations: 66
         Function evaluations: 328
         Gradient evaluations: 82
array([0.99999551, 0.99999101])
````

Reply:
Updated the output to Ubuntu 20.04 result with marked "# may vary"